fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/election/ElectionQuestion.kt

```diff
@@ -2,22 +2,33 @@ package com.github.dedis.popstellar.model.network.method.message.data.election
 
 import com.github.dedis.popstellar.model.Immutable
 import com.github.dedis.popstellar.model.objects.Election
+import com.github.dedis.popstellar.utility.MessageValidator.verify
 import com.google.gson.annotations.SerializedName
 import java.util.Collections
 import java.util.Objects
 
 @Immutable
 /** Constructor for a data Question, for the election setup */
 class ElectionQuestion(electionId: String, question: Question) {
-  val question: String = question.title
-  val id: String = Election.generateElectionQuestionId(electionId, question.title)
+  val question: String
+  val id: String
 
-  @SerializedName(value = "voting_method") val votingMethod: String = question.votingMethod
+  @SerializedName(value = "voting_method") val votingMethod: String
 
-  @SerializedName(value = "ballot_options")
-  val ballotOptions: List<String> = Collections.unmodifiableList(question.ballotOptions)
+  @SerializedName(value = "ballot_options") val ballotOptions: List<String>
 
-  @SerializedName(value = "write_in") val writeIn: Boolean = question.writeIn
+  @SerializedName(value = "write_in") val writeIn: Boolean
+
+  init {
+    // questions are already validated when constructed, but never too much validation.
+    verify().isNotEmptyBase64(electionId, "election ID").validQuestions(listOf(question))
```

Comment:
You do have your point with 'never too much validation', but in this case we're executing exactly the same check twice, as the only injection point for questions performs the check already. In my opinion we can avoid doing that but feel free to leave it as is